api/app/Console/Commands/CreateDigitalCareersPoolsCandidates.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\Pool;
+use App\Models\PoolCandidate;
+use App\Models\Classification;
+use App\Models\CmoAsset;
+use Database\Helpers\ApiEnums;
+use Illuminate\Console\Command;
+use Carbon\Carbon;
+
+class CreateDigitalCareersPoolsCandidates extends Command
+{
+    // this command updates runs through all the candidates of the old digital careers pool and creates new candidate models
+    // it works by mapping by some criteria
+    // please create the new pools first
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'create:new_digital_pool_candidates';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Map candidates of the old pool to the new pools.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        // collect starting data
+        $oldDigitalCareersId = Pool::where('key', 'ilike', 'digital_careers')->first()->toArray()['id'];
+        $candidateCollection = PoolCandidate::where('pool_id', $oldDigitalCareersId)->get();
+        $ITLevels = [1, 2, 3, 4];
+        $poolStreams = ApiEnums::poolStreams();
+        $dateNow = Carbon::now();
+        $expiryDate = Carbon::now()->addYear();
+
+        // organizing assets and streams mapping
+        $assetsCollection = CmoAsset::all();
+        global $assetArray;
+        $assetArray = [];
+        // index is asset model id field, value is asset key field
+        foreach ($assetsCollection as $asset) {
+            $assetArray[$asset->id] = $asset->key;
```

Comment:
You can use the map function to map the $assetCollection in a more functional way, just like in Javascript, and not need to use a foreach loop. https://laravel.com/docs/8.x/collections#method-map

Reply:
mapWithKeys seems to do it 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/4467/commits/fc80a132fed5eb8aeaf60d8817dfbee4ad2e0437